ipatests/test_integration/test_ipa_cert_fix.py

```diff
@@ -343,52 +343,105 @@ def install(cls, mh):
 
-        # move system date to expire cert on replica
-        tasks.move_date(self.replicas[0], 'stop', '+3years+1days')
-
-        # RA agent cert will be expired and in CA_UNREACHABLE state
-        check_status(self.replicas[0], 1, "CA_UNREACHABLE")
-
-        # renew RA agent cert
-        self.replicas[0].run_command(
-            ['ipa-cert-fix', '-v'], stdin_text='yes\n'
+        # replica operations
+        # 'Server-Cert cert-pki-ca' cert will be in CA_UNREACHABLE state
+        cmd = self.replicas[0].run_command(
+            ['getcert', 'list',
+             '-d', paths.PKI_TOMCAT_ALIAS_DIR,
+             '-n', 'Server-Cert cert-pki-ca']
+        )
+        req_id = get_certmonger_fs_id(cmd.stdout_text)
+        tasks.wait_for_certmonger_status(
+            self.replicas[0], ('CA_UNREACHABLE'), req_id, timeout=600
         )
 
-        # LDAP/HTTP/PKINIT certs will be renewed automaticaly
-        # after moving date on replica. This 3, 1 CA cert,
-        # 1 RA agent cert. Check for total 5 valid certs.
-        check_status(self.replicas[0], 5, "MONITORING")
+        # check that HTTP,LDAP,PKINIT are renewed and in MONITORING state
+        instance = realm_to_serverid(self.master.domain.realm)
+        dirsrv_cert = paths.ETC_DIRSRV_SLAPD_INSTANCE_TEMPLATE % instance
+        for cert in (paths.KDC_CERT, paths.HTTPD_CERT_FILE):
+            cmd = self.replicas[0].run_command(
+                ['getcert', 'list', '-f', cert]
+            )
+            req_id = get_certmonger_fs_id(cmd.stdout_text)
+            tasks.wait_for_certmonger_status(
+                self.replicas[0], ('MONITORING'), req_id, timeout=600
+            )
 
-        # get the req ids of all certs to renew remaining
-        # certs by re-submitting it
-        result = self.replicas[0].run_command(['getcert', 'list'])
-        req_ids = re.findall(r'\d{14}', result.stdout_text)
+        cmd = self.replicas[0].run_command(
+            ['getcert', 'list', '-d', dirsrv_cert]
+        )
+        req_id = get_certmonger_fs_id(cmd.stdout_text)
+        tasks.wait_for_certmonger_status(
+            self.replicas[0], ('MONITORING'), req_id, timeout=600
+        )
 
-        # resubmit the certs to renew them
-        for req_id in req_ids:
+        # check if replication working fine
+        testuser = 'testuser1'
+        password = 'Secret@123'
+        stdin = (f"{self.master.config.admin_password}\n"
+                 f"{self.master.config.admin_password}\n"
+                 f"{self.master.config.admin_password}\n")
+        self.master.run_command(['kinit', 'admin'], stdin_text=stdin)
+        tasks.user_add(self.master, testuser, password=password)
+        self.replicas[0].run_command(['kinit', 'admin'], stdin_text=stdin)
+        self.replicas[0].run_command(['ipa', 'user-show', testuser])
+
+        # renew shared certificates by resubmitting to certmonger
+        cmd = self.replicas[0].run_command(
```

Comment:
At this point, some of the certs may already have been renewed (the order is not garanteed).
So for each cert that is not Server-Cert cert-pki-ca, I would do the following:
- check the state and validity date. If the cert is MONITORING + valid > current date, it means it's been renewed. Nothing to do for this cert (no need to call getcert resubmit)
- if the cert is not in MONITORING or validity < current date, call getcert resubmit and wait for the cert to become MONITORING.

When all the shared certs are in MONITORING you can proceed with ipa-cert-fix.